modules/topicsFpdModule.js

```diff
@@ -73,6 +100,54 @@ export function processFpd(config, {global}, {data = topicsData} = {}) {
   });
 }
 
+// Added function addListenerToFetchTopics to listen for topic data retured from iframe loaded for SSPs(bidder)
+function addListenerToFetchTopics() {
+  window.addEventListener('message', (e) => {
+    if (e && e.data) {
+      try {
+        let data = JSON.parse(e.data);
+        if (data && data.segment && !isEmpty(data.segment.topics)) {
+          let {domain, topics, bidder} = data.segment;
+          let segmentData = getTopicsData(domain, topics);
+          let updateStore = [];
+          let bidderFoundInStore = false;
+          let storedSegments = JSON.parse(storage.getDataFromLocalStorage(`${initialTopicName}${bidder}`));
+          storedSegments?.forEach((segment) => {
+            if (segment.name === segmentData[0].name && segment.ext.segclass == segmentData[0].ext.segclass) {
+              updateStore.push(segmentData[0]);
+              bidderFoundInStore = true;
+            } else {
+              updateStore.push(segment);
+            };
+          });
+          !bidderFoundInStore && updateStore.push(segmentData[0]);
+          storage.setDataInLocalStorage(`${initialTopicName}${bidder}`, JSON.stringify(updateStore));
+        }
+      } catch (err) { }
+    }
+  });
+}
+
+function loadTopicsForBidders() {
+  addListenerToFetchTopics();
+  const topics = config.getConfig('userSync.topics');
+  if (topics) {
+    const bidders = Object.keys(topics.bidders || []);
+    bidders?.forEach((bidder) => {
+      let ifrm = document.createElement('iframe');
+      ifrm.name = `ifrm_${bidder}`;
+      ifrm.src = `${topics['bidders'][bidder]['iframeURL']}?bidder=${bidder}`;
+      ifrm.style.display = 'none';
+      window.document.documentElement.appendChild(ifrm);
+    });
+  } else {
+    logWarn(`Topics config not defined under userSync Object`);
+  }
+}
+
+// Exposing loadTopicsForBidders function in global-name-space so that API calling from domain can get the data and send it in oRTB format.
+(getGlobal()).loadTopicsForBidders = loadTopicsForBidders;
```

Comment:
Is this needed? If we are treating this the way we treat user syncs, I would just run them together. From the publisher's POV calling this seems like boilerplate.

Reply:
Removed exposing the function to global. It will be called on when topics module FPD data is processed.